src/prepare/compile.rs

```diff
@@ -0,0 +1,1239 @@
+    ins: isa::Instructions,
+    labels: Vec<(Label, Vec<isa::Reloc>)>,
+}
+
+impl Sink {
+    fn with_instruction_capacity(capacity: usize) -> Sink {
+        Sink {
+            ins: isa::Instructions::with_capacity(capacity),
+            labels: Vec::new(),
+        }
+    }
+
+    fn cur_pc(&self) -> u32 {
+        self.ins.current_pc()
+    }
+
+    fn pc_or_placeholder<F: FnOnce() -> isa::Reloc>(
+        &mut self,
+        label: LabelId,
+        reloc_creator: F,
+    ) -> u32 {
+        match self.labels[label.0] {
+            (Label::Resolved(dst_pc), _) => dst_pc,
+            (Label::NotResolved, ref mut unresolved) => {
+                unresolved.push(reloc_creator());
+                u32::max_value()
+            }
+        }
+    }
+
+    fn emit(&mut self, instruction: isa::InstructionInternal) {
+        self.ins.push(instruction);
+    }
+
+    fn emit_br(&mut self, target: Target) {
+        let Target { label, drop_keep } = target;
+        let pc = self.cur_pc();
+        let dst_pc = self.pc_or_placeholder(label, || isa::Reloc::Br { pc });
+        self.ins.push(isa::InstructionInternal::Br(isa::Target {
+            dst_pc,
+            drop_keep: drop_keep.into(),
+        }));
+    }
+
+    fn emit_br_eqz(&mut self, target: Target) {
+        let Target { label, drop_keep } = target;
+        let pc = self.cur_pc();
+        let dst_pc = self.pc_or_placeholder(label, || isa::Reloc::Br { pc });
+        self.ins
+            .push(isa::InstructionInternal::BrIfEqz(isa::Target {
+                dst_pc,
+                drop_keep: drop_keep.into(),
+            }));
+    }
+
+    fn emit_br_nez(&mut self, target: Target) {
+        let Target { label, drop_keep } = target;
+        let pc = self.cur_pc();
+        let dst_pc = self.pc_or_placeholder(label, || isa::Reloc::Br { pc });
+        self.ins
+            .push(isa::InstructionInternal::BrIfNez(isa::Target {
+                dst_pc,
+                drop_keep: drop_keep.into(),
+            }));
+    }
+
+    fn emit_br_table(&mut self, targets: &[Target], default: Target) {
+        use core::iter;
+
+        let pc = self.cur_pc();
+
+        self.ins.push(isa::InstructionInternal::BrTable {
+            count: targets.len() as u32 + 1,
+        });
+
+        for (idx, &Target { label, drop_keep }) in
+            targets.iter().chain(iter::once(&default)).enumerate()
+        {
+            let dst_pc = self.pc_or_placeholder(label, || isa::Reloc::BrTable { pc, idx });
+            self.ins
+                .push(isa::InstructionInternal::BrTableTarget(isa::Target {
+                    dst_pc,
+                    drop_keep: drop_keep.into(),
+                }));
+        }
+    }
+
+    /// Create a new unresolved label.
+    fn new_label(&mut self) -> LabelId {
+        let label_idx = self.labels.len();
+        self.labels.push((Label::NotResolved, Vec::new()));
+        LabelId(label_idx)
+    }
+
+    /// Resolve the label at the current position.
```

Comment:
doc comment goes nowhere

Reply:
Hm, I am not sure what do you mean. Looking from github, I can tell that this rustdoc belongs to the `resolve_label` function.

Reply:
non-public function so no docs for it, not really important though

Reply:
Ah, I see. It was deliberate. I am doing cause RLS actually shows you these (altthough I dont use it anymore :) )